saved-sites/saved-sites-impl/src/main/java/com/duckduckgo/savedsites/impl/sync/SavedSitesSyncPersister.kt

```diff
@@ -82,6 +74,13 @@ class SavedSitesSyncPersister @Inject constructor(
 
         if (result is Success) {
             pruneDeletedObjects()
+
+            if (conflictResolution == DEDUPLICATION) {
```

Comment:
weird to have this special case here if we have conflict resolution strategies extracted into their own classes.

Q: is this logic specific for the deduplication conflict resolution? or does this only applies to first sync (and you know the strategy on that case is deduplication)? and how do you know this is first sync?